girder/api/access.py

```diff
@@ -17,66 +17,118 @@
 #  limitations under the License.
 ###############################################################################
 
-import functools
+import six
 
-from girder.models.model_base import AccessException
 from girder.api import rest
+from girder.models.model_base import AccessException
+from girder.utility.model_importer import ModelImporter
+
+_tokenModel = ModelImporter.model('token')
 
 
-def admin(fun):
+def admin(*args, **kwargs):
     """
     Functions that require administrator access should be wrapped in this
     decorator.
-    """
-    @functools.wraps(fun)
-    def accessDecorator(*args, **kwargs):
-        rest.requireAdmin(rest.getCurrentUser())
-        return fun(*args, **kwargs)
-    accessDecorator.accessLevel = 'admin'
-    return accessDecorator
-
 
-def user(fun):
+    :param scope: To also expose this endpoint for certain token scopes,
+        pass those scopes here. If multiple are passed, all will be required.
+    :type scope: str or list of str
     """
-    Functions that allow any user (not just administrators) should be wrapped
-    in this decorator. That is, a token must be passed that has the
-    "core.user_auth" scope and a valid user ID.
+    if len(args) == 1 and callable(args[0]):  # Raw decorator
+        @six.wraps(args[0])
+        def wrapped(*iargs, **ikwargs):
+            rest.requireAdmin(rest.getCurrentUser())
+            return args[0](*iargs, **ikwargs)
+        wrapped.accessLevel = 'admin'
+        return wrapped
+    else:  # We should return a decorator
+        def dec(fun):
+            @six.wraps(fun)
+            def wrapped(*iargs, **ikwargs):
+                rest.requireAdmin(rest.getCurrentUser())
+                return fun(*iargs, **ikwargs)
+            wrapped.accessLevel = 'admin'
+            wrapped.requiredScopes = kwargs.get('scope')
```

Comment:
The ``cookie`` decorator allows either keyword or positional arguments for extra options. These new decorators expect extra options to only be keyword arguments. We should make this behavior consistent one way or another.